tests/agent/tools/test_web.py

```
@@ -1,104 +1,245 @@
+    assert not result.success
+    assert calls == ["tavily"]
+
+
+# ── provider error classification ────────────────────────────────────────
+
+
+async def test_tavily_401_classified_auth(monkeypatch):
+    async def _fake_post(self, url, **kwargs):
+        class _R:
+            status_code = 401
+            json = lambda self: {}
```

Ruff (0.16.1)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `json` as a `def`

(E731)